core/services/gasupdater/gas_updater.go

```diff
@@ -153,6 +153,10 @@ func (gu *gasUpdater) runLoop() {
 			return
 		case <-gu.mb.Notify():
```

Comment:
What is the purpose of Notify here? Won't Retrieve just block? Why not just do `case head := <- gu.mb.Retrieve()`.

Reply:
The way it works now, Retrieve never blocks and just return nil if mailbox is empty. 
Maybe for performance reasons? e.g. in many services Retrieve is in a loop, after just a single Notify